Examples/Tests/particle_thermal_boundary/analysis_2d.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+#
+# Copyright 2023-2024 Revathi Jambunathan
+#
+# This file is part of WarpX
+#
+# License: BSD-3-Clause-LBNL
+
+"""
+The script checks to see the growth in total field energy and total particle energy
+
+The input file in 2D initializes uniform plasma (electrons,ions) with
+thermal boundary condition. We do not expect the particle energy to increase
+beyond 2% in the time that it takes all particles to cross the domain boundary
+"""
+
+
+import sys
+
+import numpy as np
+
+sys.path.insert(1,'../../../../warpx/Regression/Checksum/')
+import checksumAPI
```

Comment:
I am not sure yet if we should include/exclude checksum for this test, because its very noisy.

Reply:
I would say include the checksum. In the CI test, is will always use the same random numbers and so should always get the same result.

Reply:
I agree. 
What about the random differences between the json file I generate locally, and that generated in CI? 
I will still add this checksum back and lets see the differences, and may be add a larger tolerance

Reply:
Instead of using larger tolerances, it is best to directly use the benchmark values generated by the CI.
(See https://warpx.readthedocs.io/en/latest/developers/checksum.html#reset-a-benchmark-from-the-azure-pipeline-output-on-github)

Reply:
great!!! thanks!